GoRealis has uncaught exception inside GetInstanceIDs method

Gorealis commit ID being used : `94e7c878b4e54385b6fbbb7b0c1ce5a82865af2b`

Find below the stack trace. Let me know if you need any further information.

realis-info: 2018/06/06 19:56:52 Client Error: HTTP Response code: 503
realis-info: 2018/06/06 19:56:52 Re-establishing Connection to Aurora
realis-info: 2018/06/06 19:56:52 Number of options applied to config:  4
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x60 pc=0x467cf6]

goroutine 85075208 [running]:
bytes.(*Buffer).Write(0x0, 0xc42358c000, 0x141, 0x1000, 0x0, 0x0, 0x1)
    /usr/local/go/src/bytes/buffer.go:136 +0x26
github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis/vendor/git.apache.org/thrift.git/lib/go/thrift.(*THttpClient).Write(0xc450070d40, 0xc42358c000, 0x141, 0x1000, 0x732234, 0xc455d9ce60, 0x0)
    /go/src/github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis/vendor/git.apache.org/thrift.git/lib/go/thrift/http_client.go:212 +0x4d
bufio.(*Writer).Flush(0xc450070dc0, 0x72807e, 0xc455d9ce60)
    /usr/local/go/src/bufio/bufio.go:566 +0x75
github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis/vendor/git.apache.org/thrift.git/lib/go/thrift.(*TJSONProtocol).Flush(0xc427c4c080, 0x0, 0x0)
    /go/src/github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis/vendor/git.apache.org/thrift.git/lib/go/thrift/json_protocol.go:442 +0x36
github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis/gen-go/apache/aurora.(*ReadOnlySchedulerClient).sendGetTasksWithoutConfigs(0xc455d9cf50, 0xc42827b9e0, 0xc434918bc8, 0x428e3e)
    /go/src/github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis/gen-go/apache/aurora/readonlyscheduler.go:371 +0x134
github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis/gen-go/apache/aurora.(*ReadOnlySchedulerClient).GetTasksWithoutConfigs(0xc455d9cf50, 0xc42827b9e0, 0xc434918be0, 0xc434918bd8, 0xc44a427880)
    /go/src/github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis/gen-go/apache/aurora/readonlyscheduler.go:346 +0x39
github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis.(*realisClient).GetInstanceIds.func1(0xc400000008, 0xda03a0, 0xc4247f7b08)
    /go/src/github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis/realis.go:474 +0x3a
github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis.(*realisClient).thriftCallWithRetries.func1(0xc438f5c140, 0xc434918ce8, 0xc434918d58, 0xc434918e28)
    /go/src/github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis/retry.go:150 +0x6b
github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis.(*realisClient).thriftCallWithRetries(0xc438f5c140, 0xc434918e28, 0x2b, 0xc44700dbf0, 0x1)
    /go/src/github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis/retry.go:153 +0x222
github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis.(*realisClient).GetInstanceIds(0xc438f5c140, 0xc424b6a8d0, 0xc420017d40, 0xc434918f00, 0x4434e7, 0xc424b19756)
    /go/src/github.paypal.com/PaaS-R/Genesis-Recon/vendor/github.com/paypal/gorealis/realis.go:475 +0x198
github.paypal.com/PaaS-R/Genesis-Recon/aurora.retrieveTaskConfig(0x11be740, 0xc438f5c140, 0xc424b6a8d0, 0xd68688, 0x2, 0xc424b19740)
    /go/src/github.paypal.com/PaaS-R/Genesis-Recon/aurora/queryAurora.go:168 +0x62
github.paypal.com/PaaS-R/Genesis-Recon/aurora.GetAuroraData(0xc43e947720, 0x0, 0xc42035d920)
    /go/src/github.paypal.com/PaaS-R/Genesis-Recon/aurora/queryAurora.go:71 +0x150
main.checkForDataDiscrepency(0xc422443ee0, 0x12, 0xc422442680, 0x11, 0xc43e947720, 0xc442a8cd30, 0xc422b62000, 0xc422b620c0, 0xc42035d920, 0xc42035d9e0, ...)
    /go/src/github.paypal.com/PaaS-R/Genesis-Recon/main.go:584 +0x2f2c
created by main.beginOperation
    /go/src/github.paypal.com/PaaS-R/Genesis-Recon/main.go:254 +0x2327

--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -2,10 +2,11 @@
 
 
 [[projects]]
+  branch = "0.10.0-http-client-fix"
   name = "git.apache.org/thrift.git"
   packages = ["lib/go/thrift"]
-  revision = "b2a4d4ae21c789b689dd162deb819665567f481c"
-  version = "0.10.0"
+  revision = "cb1afec972a85791e9b24a04b60fc9dbbfc3cda3"
+  source = "github.com/rdelval/thrift"
 
 [[projects]]
   name = "github.com/davecgh/go-spew"
@@ -38,6 +39,6 @@
 [solve-meta]
   analyzer-name = "dep"
   analyzer-version = 1
-  inputs-digest = "6b1d8788979382ad684db5a62850b9d206014ea36f020a0a481331adf3c234dd"
+  inputs-digest = "fdb631400420ca7299ad56b66175c9157ca073e3de52f666a84fc1d6fa893978"
   solver-name = "gps-cdcl"
   solver-version = 1

--- a/Gopkg.toml
+++ b/Gopkg.toml
@@ -1,6 +1,7 @@
 [[constraint]]
     name = "git.apache.org/thrift.git"
-    version = "0.10.0"
+    branch = "0.10.0-http-client-fix"
+    source = "github.com/rdelval/thrift"
 
 [[constraint]]
     name = "github.com/pkg/errors"

